### PR TITLE
Feature/vesting switch chain tab

### PR DESCRIPTION
**Pull request by raman-pelith**
image
image

clickup: CU-31zrauw

### CODE REVIEW

File: src/hooks/vesting/useVestingInfo.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export type VestingInfoType = {
+  [chainId in ChainId]?: {vestingValue?: BigNumber, vestingProportion?: BigNumber, lastWithdrawalTime?: BigNumber};
+};
+
+export default function useVestingInfo(): {
+  vestingInfo: VestingInfoType;
+} {
+  const [vestingInfo, setVestingInfo] = useState<VestingInfoType>();
+
+  const {fetchVestingInfoResult: mainnetVestingInfo} = useFetchVestingInfo(ChainId.MAINNET);
```
Comment (hung-pelith) [inline src/hooks/vesting/useVestingInfo.ts:15]: blanks in side {} 🥺

Comment (raman-pelith) [reply src/hooks/vesting/useVestingInfo.ts:15]: sorry about that QQ

**Comment by ytwang00** [comment]
Task linked: [CU-31zrauw Add tabs to vesting page for cross chain](https://app.clickup.com/t/31zrauw)

File: src/components/VestingPage/index.tsx [hunk 2]
```diff
@@ -5,66 +5,84 @@ import {
...
-    vestingContract,
-    'lastWithdrawalTime',
-    [account],
+  const isChainSupported = vestingSupportChainIdSet.has(chainId);
+  const [activeChainTab, setActiveChainTab] = useState(
+    isChainSupported ? chainId : ChainId.MAINNET
   );
+
+  useEffect(() => {
+    if (vestingSupportChainIdSet.has(chainId)) {
```
Comment (ytwang00) [inline src/components/VestingPage/index.tsx:57]: vestingSupportChainIdSet.has(chainId) => isChainSupported

File: src/hooks/vesting/useVestingInfo.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+} {
+  const [vestingInfo, setVestingInfo] = useState<VestingInfoType>();
+
+  const { fetchVestingInfoResult: mainnetVestingInfo } = useFetchVestingInfo(ChainId.MAINNET);
+  const { fetchVestingInfoResult: bscVestingInfo } = useFetchVestingInfo(ChainId.BSC);
+  const { fetchVestingInfoResult: fantomVestingInfo } = useFetchVestingInfo(ChainId.FANTOM);
+  const { fetchVestingInfoResult: polygonVestingInfo } = useFetchVestingInfo(ChainId.POLYGON);
+  const { fetchVestingInfoResult: kovanVestingInfo } = useFetchVestingInfo(ChainId.KOVAN);
+
+  useEffect(() => {
```
Comment (ytwang00) [inline src/hooks/vesting/useVestingInfo.ts:21]: I think u could add a debounce for this

Comment (ytwang00) [reply src/hooks/vesting/useVestingInfo.ts:21]: setVestingResult = () => {
setVestingInfo(Object.fromEntries([
      mainnetVestingInfo || [[ChainId.MAINNET], undefined], 
      bscVestingInfo || [[ChainId.BSC], undefined], 
      fantomVestingInfo || [[ChainId.FANTOM], undefined], 
      polygonVestingInfo || [[ChainId.POLYGON], undefined], 
      kovanVestingInfo || [[ChainId.KOVAN], undefined]
    ]));}
throttleFunc = throttle(setVestingResult)

Comment (ytwang00) [reply src/hooks/vesting/useVestingInfo.ts:21]: 大概像這樣 其他的param你補一下

**Review by ytwang00 — APPROVED** [review]

**Review by ytwang00 — APPROVED** [review]

**Pull request merged**